specs/networking/libp2p-standardization.md

```diff
@@ -0,0 +1,192 @@
+	beacon blocks to all nodes on the networks.
+- The `beacon_attestation` topic - This topic is used to propagate
+	aggregated attestations to subscribing nodes (typically block proposers) to
+	be included into future blocks. Attestations will be aggregated in their
+	respective subnets before publishing on this topic.
+
+Shards will be grouped into their own subnets (defined by a shard topic). The
+number of shard subnets will be defined via `SHARD_SUBNET_COUNT` and the shard
+`shard_number % SHARD_SUBNET_COUNT` will be assigned to the topic:
+`shard{shard_number % SHARD_SUBNET_COUNT}`.
+
+### Messages
+
+#### Libp2p Specification
+
+*This section simply outlines the data sent across the wire as specified by
+libp2p - this section is aimed at gossipsub implementers to standardize their implementation of this protocol*
+
+Libp2p raw gossipsub messages are sent across the wire as fixed-size length-prefixed byte arrays.
+
+The byte array is prefixed with an unsigned 64 bit length number encoded as an
+`unsigned varint` (https://github.com/multiformats/unsigned-varint). Gossipsub messages therefore take the form:
+```
++--------------------------+
+|     message length       |
++--------------------------+
+|                          |
+|       body (<1M)         |
+|                          |
++--------------------------+
+```
+
+The body represents a protobuf-encoded [Message](https://github.com/libp2p/go-libp2p-pubsub/blob/master/pb/rpc.proto#L17-L24). 
+
+In the following section we discuss the data being sent in the `data` field of
+the protobuf gossipsub `Message`.
+
+#### Eth2.0 Specifics
+
+Each message has a maximum size of 512KB (estimated from expected largest uncompressed
+block size).
+
+The `data` that is sent in a Gossipsub message is an SSZ-encoded object. For the `beacon_block` topic,
+this will be a `beacon_block`. For the `beacon_attestation` topic, this will be
+an `attestation`.
+
+## Eth-2 RPC
+
+#### Protocol Id: `/eth/serenity/beacon/rpc/1`
+
+The [RPC Interface](./rpc-interface.md) is specified in this repository.
+
+## Identify
+
+#### Protocol Id: `/ipfs/id/1.0.0` (to be updated to `/p2p/id/1.0.0`)
+
+*To be updated to incorporate discv5*
+
+
+The Identify protocol (defined in go - [identify-go](https://github.com/ipfs/go-ipfs/blob/master/core/commands/id.go) and rust [rust-identify](https://github.com/libp2p/rust-libp2p/blob/master/protocols/identify/src/lib.rs))
+allows a node A to query another node B which information B knows about A. This also includes the addresses B is listening on.
+
+This protocol allows nodes to discover addresses of other nodes to be added to
+peer discovery. It further allows nodes to determine the capabilities of all it's connected
+peers.
+
+### Configuration Parameters
+
+The protocol has two configurable parameters, which can be used to identify the
+type of connecting node. Suggested format:
+```
+	version: `/eth/serenity/1.0.0`
```

Comment:
👍

Reply:
Ok, to my knowledge, this is currently not the case in rust-libp2p. I'll make a PR to make these configurable.